bust-a-move-framework/src/main/java/bustamove/game/Highscore.java

```diff
@@ -0,0 +1,215 @@
+     */
+    private void addEntry(final String entry) {
+        String[] parts = entry.split(",");
+        if (parts.length != 2) {
+            Log.log("High score entry has wrong parameters.");
+        }
+        int score = 0;
+        String name = parts[0];
+        try {
+            score = Integer.parseInt(parts[1]);
+        } catch (NumberFormatException e) {
+            Log.log("Error loading high score entry: [" + entry
+                    + "], score is not an int.");
+        }
+        addEntry(name, score);
+    }
+
+    /**
+     * Adds a high score entry.
+     * @param name The player's name.
+     * @param score The score they achieved.
+     */
+    private void addEntry(final String name, final int score) {
+        if (!highscores.containsKey(score)) {
+            highscores.put(score, new ArrayList<String>());
+        }
+        highscores.get(score).add(name);
+    }
+
+    /**
+     * Adds and entry to high scores and saves the highscores to a file.
+     * @param name The player's name
+     * @param score The score they achieved.
+     */
+    public void addEntryAndSave(final String name, final int score) {
+        addEntry(name, score);
+        save();
+    }
+
+    /**
+     * Saves the high scores to the specified file.
+     */
+    private void save() {
+        PrintWriter writer = null;
+        try {
+            writer = new PrintWriter(FILE, "UTF-8");
+            Entry<Integer, ArrayList<String>> entry = highscores.lastEntry();
+            int saved = 0;
+            while (entry != null && saved < GameConfig.HIGHSCORE_ENTRIES) {
+                saved += saveEntry(writer, entry);
+                entry = highscores.lowerEntry(entry.getKey());
+            }
+            writer.close();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Saves an TreeMap entry to a file.
+     * @param writer The PrintWriter to write the file to.
+     * @param entry The entry to write to the file
+     * @return an int with how many entries where saved to the file.
+     */
+    private int saveEntry(final PrintWriter writer,
+            final Entry<Integer, ArrayList<String>> entry) {
+        int added = 0;
+        for (String name : entry.getValue()) {
+            if (saveEntry(writer, name, entry.getKey())) {
+                added++;
+            }
+        }
+        return added;
+    }
+
+    /**
+     * Saves an entry to a file.
+     * @param writer The printwriter to write to.
+     * @param name The name of the player.
+     * @param score The score they achieved.
+     * @return A boolean for if it was successfully saved (true)
+     * and false otherwise.
+     */
+    private boolean saveEntry(final PrintWriter writer,
+            final String name, final int score) {
+        writer.println(name + "," + score);
+        return true;

```

Comment:
Do you need a separate method for this? This will always return true unless an exception occurs(which is caught in the save method.)